homeassistant/components/sia/hub.py

```diff
@@ -0,0 +1,197 @@
+            device_registry.async_get_or_create(
+                config_entry_id=self.entry_id,
+                identifiers={(DOMAIN, port, account)},
+                name=f"{port} - {account}",
+            )
+        self.shutdown_remove_listener = self._hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_STOP, self.async_shutdown
+        )
+        self._reactions = load_json("homeassistant/components/sia/reactions.json")
+
+    async def async_shutdown(self, _: Event):
+        """Shutdown the SIA server."""
+        await self.sia_client.stop()
+
+    def _create_sensors(self):
+        """Create all the sensors."""
+        for zone in self._zones:
+            ping = self._get_ping_interval(zone[CONF_ACCOUNT])
+            for entity_type in zone[CONF_SENSORS]:
+                entity_id, entity_name = self._get_entity_id_and_name(
+                    zone[CONF_ACCOUNT], zone[CONF_ZONE], entity_type
+                )
+                self.states[entity_id] = SIAAlarmControlPanel(
+                    entity_id,
+                    entity_name,
+                    self._port,
+                    zone[CONF_ACCOUNT],
+                    zone[CONF_ZONE],
+                    ping,
+                    self._hass,
+                )
+
+    def _get_entity_id_and_name(
+        self, account: str, zone: int = 0, entity_type: str = None
+    ) -> Tuple[str, str]:
+        """Give back a entity_id and name according to the variables."""
+        if zone == 0:
+            return (
+                self._get_entity_id(account, zone, entity_type),
+                f"{self._port} - {account} - Last Heartbeat",
+            )
+        if entity_type:
+            return (
+                self._get_entity_id(account, zone, entity_type),
+                f"{self._port} - {account} - zone {zone} - {entity_type}",
+            )
+        return None
+
+    def _get_zones(self, account: str, zone=None) -> list:
+        """Get a list of zones for a account."""
+        if zone and int(zone) > 0:
+            return [zone]
+        return [
+            item[CONF_ZONE] for item in self._zones if item[CONF_ACCOUNT] == account
+        ]
+
+    def _get_entity_id(
+        self, account: str, zone: int = 0, entity_type: str = None
+    ) -> str:
+        """Give back a entity_id according to the variables, defaults to the hub sensor entity_id."""
+        if zone == 0 or entity_type == DEVICE_CLASS_TIMESTAMP:
+            return f"{self._port}_{account}_{HUB_SENSOR_NAME}"
+        if entity_type:
+            return f"{self._port}_{account}_{zone}_{entity_type}"
+        return None
+
+    def _get_ping_interval(self, account: str) -> timedelta:
+        """Return the ping interval for specified account."""
+        for acc in self._accounts:
+            if acc[CONF_ACCOUNT] == account:
+                return timedelta(minutes=acc[CONF_PING_INTERVAL])
+        return None
+
+    async def update_states(self, event: SIAEvent):
```

Comment:
Does this mean it's event driven? Should the entity set `should_poll` to `False`?

Reply:
It is, I will fix that.

Reply:
It is indeed event driven, I had a _should_poll attribute, fixed that.